include/mutex.h

```diff
@@ -0,0 +1,37 @@
+#ifndef _MUTEX_
+#define _MUTEX_
+
+#include <atomic>
+
+//! Hybrid SpinMutex class
+//! \brief SpinMutex class that offers a fast locking and unlocking mechanism
+class SpinMutex {
```

Comment:
Do you want to call it Hybrid or Adaptive Mutex instead? I think it's clearer, but I am also OK with SpinMutex

Reply:
It's a spin lock and a Mutex, hence SpinMutex

Reply:
Understood